flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/over/NonTimeRangeUnboundedPrecedingFunction.java

```diff
@@ -168,127 +91,19 @@ public NonTimeRangeUnboundedPrecedingFunction(
-            sortedListStateDescriptor.enableTimeToLive(ttlConfig);
-        }
-        sortedListState = getRuntimeContext().getState(sortedListStateDescriptor);
-
-        // Initialize state which maintains the actual row
-        valueStateDescriptor =
-                new MapStateDescriptor<Long, RowData>(
-                        "valueMapState", Types.LONG, inputRowTypeInfo);
-        if (ttlConfig.isEnabled()) {
-            valueStateDescriptor.enableTimeToLive(ttlConfig);
-        }
-        valueMapState = getRuntimeContext().getMapState(valueStateDescriptor);
-
-        // Initialize accumulator state per row
-        InternalTypeInfo<RowData> accTypeInfo = InternalTypeInfo.ofFields(accTypes);
-        accStateDescriptor =
-                new MapStateDescriptor<RowData, RowData>(
-                        "accMapState", sortKeyRowTypeInfo, accTypeInfo);
-        if (ttlConfig.isEnabled()) {
-            accStateDescriptor.enableTimeToLive(ttlConfig);
-        }
-        accMapState = getRuntimeContext().getMapState(accStateDescriptor);
-
-        // metrics
-        this.numOfIdsNotFound =
-                getRuntimeContext().getMetricGroup().counter(IDS_NOT_FOUND_METRIC_NAME);
-        this.numOfSortKeysNotFound =
-                getRuntimeContext().getMetricGroup().counter(SORT_KEYS_NOT_FOUND_METRIC_NAME);
-    }
-
-    /**
-     * Puts an element from the input stream into state or removes it from state if the input is a
-     * retraction. Emits the aggregated value for the newly inserted element and updates all results
-     * that are affected by the added or removed row. Emits the same aggregated value for all
-     * elements with the same sortKey to comply with the sql RANGE syntax.
-     *
-     * @param input The input value.
-     * @param ctx A {@link Context} that allows querying the timestamp of the element and getting
-     *     TimerService for registering timers and querying the time. The context is only valid
-     *     during the invocation of this method, do not store it.
-     * @param out The collector for returning result values.
-     * @throws Exception
-     */
-    @Override
-    public void processElement(
-            RowData input,
-            KeyedProcessFunction<K, RowData, RowData>.Context ctx,
-            Collector<RowData> out)
-            throws Exception {
-        RowKind rowKind = input.getRowKind();
-
-        switch (rowKind) {
-            case INSERT:
-            case UPDATE_AFTER:
-                insertIntoSortedList(input, out);
-                break;
-
-            case DELETE:
-            case UPDATE_BEFORE:
-                removeFromSortedList(input, out);
-                break;
-        }
-
-        // Reset acc state since we can have out of order inserts into the ordered list
-        aggFuncs.resetAccumulators();
-        aggFuncs.cleanup();
+            RowDataKeySelector sortKeySelector,
+            InternalTypeInfo<RowData> accKeyRowTypeInfo) {
```

Comment:
No need to pass this in to the function.
For the RANGE operator this should be the `sortKeyRowTypeInfo` which can be derived from the other args.

Reply:
Removed the last parameter `InternalTypeInfo<RowData> accKeyRowTypeInfo`